askama/src/lib.rs

```diff
@@ -55,20 +55,53 @@
 //!   specified extension (`ext`): if the extension is `html`, `htm` or `xml`,
 //!   the `html` escape mode is used; otherwise, no implicit escaping is done.
 //!   Setting an escape mode explicitly overrides the inferred value.
+//! * `syntax` (as `syntax = "foo"`): set the syntax name for a parser defined
+//!   in the configuration file. The default syntax , "default",  is the one
+//!   provided by Jinja.
 //!
 //! ## Configuration
 //!
 //! At compile time, Askama will read optional configuration values from
 //! `askama.toml` in the crate root (the directory where `Cargo.toml` can
-//! be found). Currently, this only covers the directories to search for
-//! templates.
+//! be found). Currently, this covers the directories to search for templates,
+//! as well as the personalization of syntax.
+//!
 //!
 //! This example file demonstrates the default configuration:
 //!
 //! ```toml
 //! [general]
 //! # Directories to search for templates, relative to the crate root.
 //! dirs = ["templates"]
+//!
+//! # Default syntax name
+//! default_syntax = "default"
+//!
+//! ```
+//!
+//! Syntax consists in at least the attribute `name` which has to be unique for all syntax.
+//! Delimiters can be added to personalized syntax for example: `block_start`, `block_end`,
+//! `comment_start`, `comment_end`, `expr_start` and `expr_end`. Delimiters must have exactly 2
+//! characters and that ones ending in `*_start` must have one character in common in the same
```

Comment:
"must be exactly 2 characters long and start delimiters must all start with the same character."

Reply:
must be exactly 2 characters long and start delimiters must all start **or end** with the same character.

Reply:
Why do we have that requirement, anyway? Is it that start delimiters must all start, and end delimiters must all end with the same character?

Reply:
It's because of the take_content function. Only in the start. Because it checks the input 1 to 1 and 2 correct ones are needed. Then the https://github.com/djc/askama/blob/master/askama_derive/src/parser.rs#L137-L150 can fail or give false positives or not parse them.